src/main/java/net/dries007/tfc/objects/fluids/FluidsTFC.java

```diff
@@ -162,7 +179,24 @@ public static void registerFluids()
                 TANNIN = registerFluid(new Fluid("tannin", STILL, FLOW, 0xFF63594E)),
                 LIMEWATER = registerFluid(new Fluid("limewater", STILL, FLOW, 0xFFB4B4B4)),
                 CURDLED_MILK = registerFluid(new Fluid("milk_curdled", STILL, FLOW, 0xFFFFFBE8)),
-                MILK_VINEGAR = registerFluid(new Fluid("milk_vinegar", STILL, FLOW, 0xFFFFFBE8))
+                MILK_VINEGAR = registerFluid(new Fluid("milk_vinegar", STILL, FLOW, 0xFFFFFBE8)),
+                //dye block
+                WHITE_DYE = registerFluid(new Fluid("white_dye", STILL, FLOW, 0xFFF9FFFE)),
+                ORANGE_DYE = registerFluid(new Fluid("orange_dye", STILL, FLOW, 0xFFF9801D)),
+                MAGENTA_DYE = registerFluid(new Fluid("magenta_dye", STILL, FLOW, 0xFFC74EBD)),
+                LIGHT_BLUE_DYE = registerFluid(new Fluid("light_blue_dye", STILL, FLOW, 0xFF3AB3DA)),
+                YELLOW_DYE = registerFluid(new Fluid("yellow_dye", STILL, FLOW, 0xFFFED83D)),
+                LIME_DYE = registerFluid(new Fluid("lime_dye", STILL, FLOW, 0xFF80C71F)),
+                PINK_DYE = registerFluid(new Fluid("pink_dye", STILL, FLOW, 0xFFF38BAA)),
+                GRAY_DYE = registerFluid(new Fluid("gray_dye", STILL, FLOW, 0xFF474F52)),
+                LIGHT_GRAY_DYE = registerFluid(new Fluid("light_gray_dye", STILL, FLOW, 0xFF9D9D97)),
+                CYAN_DYE = registerFluid(new Fluid("cyan_dye", STILL, FLOW, 0xFF169C9C)),
+                PURPLE_DYE = registerFluid(new Fluid("purple_dye", STILL, FLOW, 0xFF8932B8)),
+                BLUE_DYE = registerFluid(new Fluid("blue_dye", STILL, FLOW, 0xFF3C44AA)),
+                BROWN_DYE = registerFluid(new Fluid("brown_dye", STILL, FLOW, 0xFF835432)),
+                GREEN_DYE = registerFluid(new Fluid("green_dye", STILL, FLOW, 0xFF5E7C16)),
+                RED_DYE = registerFluid(new Fluid("red_dye", STILL, FLOW, 0xFFB02E26)),
+                BLACK_DYE = registerFluid(new Fluid("black_dye", STILL, FLOW, 0xFF1D1D21))
```

Comment:
This can be replaced with a single map constructor.
 - One niche thing is that `EnumDyeColor#getColorValue` is client only, but `getColorComponentValues` is not. I think we can solve this via `new Color(f, g, b).getRGB()`

```java
DYE_FLUIDS = Arrays.stream(EnumDyeColor.values()).collect(Collectors.toMap(
    color -> color,
    color -> {
        float[] c = color.getColorComponentValues();
        return registerFluid(new Fluid(color.getName() + "_dye", STILL, FLOW, new Color(c[0], c[1], c[2]).getRGB()));
));
```